File: Competetive programming/Python/Unit 4/Longest_Nap.py
print("===== LONGEST NAP =====\n")

appointments = int(input("Enter number of meetings: "))

times = []

for i in range(appointments):

    start = int(input(f"Meeting {i+1} Start Hour: "))
    end = int(input(f"Meeting {i+1} End Hour: "))

    times.append((start, end))

times.sort()

longest = 0

for i in range(len(times)-1):

    gap = times[i+1][0] - times[i][1]

    if gap > longest:
        longest = gap

print("\nLongest Nap =", longest, "hours")

File: Competetive programming/C++/Unit 4/Longest_Nap.cpp
#include <iostream>
using namespace std;

int main() {

    int n;

    cout << "===== LONGEST NAP =====\n\n";

    cout << "Enter number of appointments: ";
    cin >> n;

    int start[100], end[100];

    cout << "\nEnter appointments in 24-hour format.\n";
    cout << "Example: 10 11 (means 10:00 AM to 11:00 AM)\n\n";

    for(int i = 0; i < n; i++) {

        cout << "Appointment " << i + 1 << " Start Hour: ";
        cin >> start[i];

        cout << "Appointment " << i + 1 << " End Hour: ";
        cin >> end[i];
    }

    int longest = start[0] - 10;

    for(int i = 0; i < n - 1; i++) {

        int gap = start[i + 1] - end[i];

        if(gap > longest)
            longest = gap;
    }

    int lastGap = 18 - end[n - 1];

    if(lastGap > longest)
        longest = lastGap;

    cout << "\nLongest Nap Duration = "
         << longest
         << " hour(s)\n";

    return 0;
}